swaybar/bar.c

```diff
@@ -537,6 +537,7 @@ static void free_seats(struct wl_list *list) {
 void bar_teardown(struct swaybar *bar) {
 #if HAVE_TRAY
 	destroy_tray(bar->tray);
+	sd_bus_unref(bar->tray->bus);
```

Comment:
Doesn't this cause a use-after-free?

Sounds cleaner to move `sd_bus_unref()` into `destroy_tray()`.

Reply:
Yes and also destroy_tray already calls `sd_bus_flush_close_unref` so I removed this line.